### PR TITLE
[bdd-engine] Add story sorting by a numeric value

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/order/StoryOrder.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.Comparator;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.jbehave.core.model.Story;
+
+/**
+ * Methods of this class are expected to be used in test projects as static factory methods for story comparators
+ */
+public final class StoryOrder
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/order/StoryOrder.java:30]: `MetaBasedStoryExecutionPriority`

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/order/StoryOrder.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private StoryOrder()
+    {
+    }
+
+    /**
+     * Create a comparator that sorts stories by a numeric value extracted by the key in descending order
+     * @param key key to extract a numeric meta value, must not be a blank
+     * @return comparator
+     */
+    public static Comparator<Story> byNumericValue(String key)
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/order/StoryOrder.java:41]: `public static Comparator<Story> byNumericMetaValue(String metaName)`

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/order/StoryOrder.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        isTrue(StringUtils.isNotBlank(key), "The key must be a non-zero length string");
+        return Comparator.comparing(story -> getNumericValue(story, key), Comparator.reverseOrder());
+    }
+
+    private static int getNumericValue(Story story, String key)
+    {
+        String priorityAsStrinig = story.getMeta().getProperty(key);
+        if (priorityAsStrinig.isEmpty())
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/order/StoryOrder.java:50]: I don't think an empty value should be considered as valid

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/order/StoryOrder.java (hunk 1 above)
Comment (uarlouski) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/order/StoryOrder.java]: renamed, but here is a risk that we will want to add some more ordering strategies and will have to rename this class to something more generic leading to difficulties in support

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/priority/MetaBasedStoryExecutionPriority.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    /**
+     * Create a comparator that sorts stories by a numeric meta value extracted by the meta name in descending order
+     * @param metaName meta name to extract a numeric meta value, must not be a blank
+     * @return comparator
+     */
+    public static Comparator<Story> byNumericMetaValue(String metaName)
+    {
+        isTrue(StringUtils.isNotBlank(metaName), "The meta name must be a non-zero length string");
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/priority/MetaBasedStoryExecutionPriority.java:44]: `isNotBlank ` also checks for whitespace-only string

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/priority/MetaBasedStoryExecutionPriority.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    /**
+     * Create a comparator that sorts stories by a numeric meta value extracted by the meta name in descending order
+     * @param metaName meta name to extract a numeric meta value, must not be a blank
+     * @return comparator
+     */
+    public static Comparator<Story> byNumericMetaValue(String metaName)
+    {
+        isTrue(metaName.matches("[^ ]+"), "The meta name must be a non-zero length string without spaces");
```
Comment (ikalinin1) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/priority/MetaBasedStoryExecutionPriority.java:43]: StringUtils.isNotEmpty ?

Comment (uarlouski) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/priority/MetaBasedStoryExecutionPriority.java:43]: values with spaces are not allowed

**Review by valfirst — APPROVED** [review]

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/priority/MetaBasedStoryExecutionPriority.java (hunk 5 above)
Comment (ikalinin1) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/priority/MetaBasedStoryExecutionPriority.java:43]: Yep, I got this, why not check it's a number then?

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**